### PR TITLE
feat: ✨ Add rivers in city mode

### PR DESCRIPTION
**Pull request by JulesL2**
Add rivers in city mode

## Description

Add the rivers to the city poster

See https://github.com/ThomasParistech/pretty-gpx/pull/9 for the discussion

## Proof of Functionality

rivers

### CODE REVIEW

File: pretty_gpx/common/data/overpass_processing.py [hunk 1]
```diff
@@ -20,3 +29,193 @@ def get_ways_coordinates_from_results(api_result: overpy.Result) -> list[ListLon
...
+    """
+    Get the members from a relation and classify them by their role
+    """
+    relation_members = relation.members
+    if relation_members is not None:
+        outer_geometry_l = []
+        inner_geometry_l = []
+        for member in relation_members:
+            if type(member) == overpy.RelationRelation:
+                logger.info(f"Found a RelationRelation i.e. a relation in the members of a relation")
```
Comment (ThomasParistech) [inline pretty_gpx/common/data/overpass_processing.py:98]: too verbose

File: pretty_gpx/city/data/rivers.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+from pretty_gpx.common.data.overpass_processing import get_polygons_from_relations
+from pretty_gpx.common.data.overpass_processing import get_polygons_from_closed_ways
+
+
+
+
+RIVERS_CACHE = GpxDataCacheHandler(name='rivers', extension='.pkl')
+
+
+def download_city_rivers(bounds: GpxBounds) -> Surface_Polygons:
```
Comment (ThomasParistech) [inline pretty_gpx/city/data/rivers.py:27]: is the interior polygon part of the river actually used ?

Comment (JulesL2) [reply pretty_gpx/city/data/rivers.py:27]: Yes, there are some islands on the seine river 😀

File: pretty_gpx/common/data/overpass_processing.py (hunk 1 above)
Comment (JulesL2) [reply pretty_gpx/common/data/overpass_processing.py:98]: Yes, I will change to debug

File: pretty_gpx/city/data/rivers.py (hunk 2 above)
Comment (JulesL2) [reply pretty_gpx/city/data/rivers.py:27]: This feature allows the code to draw the 2 branches on the river on multiple locations

File: .devcontainer/requirements.txt [hunk 3]
```diff
@@ -15,4 +15,6 @@ pathvalidate==3.2.0
 svgpathtools==1.6.1
```
Comment (ThomasParistech) [inline .devcontainer/requirements.txt:15]: I've updated the PR validator github action to handle forked repos
can you rebase your branch on the latest main ?

Comment (JulesL2) [reply .devcontainer/requirements.txt:15]: It works now 🥳

**Comment by JulesL2** [comment]
@ThomasParistech All comments taken into account !

**Review by ThomasParistech — APPROVED** [review]
:+1:

**Pull request merged**